frappe/desk/page/setup_wizard/setup_wizard.js

```diff
@@ -213,7 +213,11 @@ frappe.setup.SetupWizard = class SetupWizard extends frappe.ui.Slides {
 
 	abort_setup(fail_msg) {
 		this.$working_state.find(".state-icon-container").html("");
-		fail_msg = fail_msg ? fail_msg : __("Failed to complete setup");
+		fail_msg = fail_msg
+			? fail_msg
+			: frappe.last_response.setup_wizard_failure_message
+			? frappe.last_response.setup_wizard_failure_message
```

Comment:
FYI last response can be somewhat unreliable, but during setup I am assuming nothing else is going on, so it should be fine.

Reply:
Ah okay
I was trying to rely on the error callback, but that never seemed to have the value I would've expected it to